clang/test/CIR/CodeGen/binop.cpp

```diff
@@ -23,4 +23,32 @@ void b0(int a, int b) {
 // CHECK: = cir.binop(shl, %21, %22) : i32
 // CHECK: = cir.binop(and, %24, %25) : i32
 // CHECK: = cir.binop(xor, %27, %28) : i32
-// CHECK: = cir.binop(or, %30, %31) : i32
\ No newline at end of file
+// CHECK: = cir.binop(or, %30, %31) : i32
+
+void b1(bool a, bool b) {
+  bool x = a && b;
+  x = x || b;
+}
+
```

Comment:
Your original patch was handling some casts/conversions between int and bool. For instance, I noticed this doesn't work yet:
```
bool b1(unsigned a, unsigned b) {
  unsigned x = a && b;
  x = x || b;
  return x;
}
```
Do you plan to handle that in another PR? No need to do it right now, just wondering if you have a plan and making sure the proper asserts/unreachables are added in this PR so we can incrementally make progress later.

Reply:
I will send another patch that implements `CreateZExtOrBitCast`.